aggregate_spatial_window.json

```diff
@@ -0,0 +1,98 @@
+    "description": "Aggregates statistics over the horizontal spatial dimensions (axes `x` and `y`) of the data cube.\n\nThe pixel grid for the axes `x` and `y` is divided into non-overlapping windows with the size specified in the parameter `size`. If the number of values for the axes `x` and `y` is not a multiple of the window size, the behaviour specified in `boundary` parameter is applied.\nFor each of these windows, the reducer process computes the result.",
+    "categories": [
+        "cubes",
+        "aggregate & resample"
+    ],
+    "parameters": [
+        {
+            "name": "data",
+            "description": "A raster data cube.",
+            "schema": {
+                "type": "object",
+                "subtype": "raster-cube"
+            }
+        },
+        {
+            "name": "reducer",
+            "description": "A reducer to be applied on all values of each geometry. A reducer is a single process such as ``mean()`` or a set of processes, which computes a single value for a list of values, see the category 'reducer' for such processes.",
+            "schema": {
+                "type": "object",
+                "subtype": "process-graph",
+                "parameters": [
+                    {
+                        "name": "data",
+                        "description": "An array with elements of any type.",
+                        "schema": {
+                            "type": "array",
+                            "items": {
+                                "description": "Any data type."
+                            }
+                        }
+                    },
+                    {
+                        "name": "context",
+                        "description": "Additional data passed by the user.",
+                        "schema": {
+                            "description": "Any data type."
+                        },
+                        "optional": true,
+                        "default": null
+                    }
+                ]
+            }
+        },
+        {
+            "name": "size",
+            "description": "Window sizes in pixels along the horizontal spatial dimensions.\n\nThe first value corresponds to the `x` axis, the second values corresponds to the `y` axis.",
+            "schema": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "subtype": "chunk-size",
+                    "required": [
+                        "value"
+                    ],
+                    "properties": {
+                        "value": {
+                            "type": "number",
+                            "minimum": 1,
+                            "description": "The unit the values are given in pixels (`px`)"
+                        }
+                    }
+                }
+            }
+        },
+        {
+            "name": "boundary",
```

Comment:
Open questions: What is trimmed? Where is the padding added? Equally on each side? At the "end"? At the "beginning"? See https://github.com/Open-EO/openeo-processes/pull/192#discussion_r490122368

@clausmichele 
> Good question. Actually I'm using xarray coarsen method, which has also the option to align top-left or top-right the windows http://xarray.pydata.org/en/v0.15.1/generated/xarray.DataArray.coarsen.html
So, if aligned to the left it will trim the excess on the right and vice versa.

@m-mohr 
>  I'll put this into proposal stage first (see #196) and then we can elaborate during development whether a paramter should be added. For now we should just specify a default, either upper-left or lower-right. I think I'd default to pad the lower-right.

Reply:
I would say that pad or trim on the right/bottom could be default, with the window aligned top-left at the beginning.

Reply:
Sorry for leaving this open so long. I've now added a parameter `align` as it is available in the resample process. This should now clarify all alignment issues. Please review the changes.